akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -59,19 +59,31 @@ private[projection] object R2dbcOffsetStore {
   final case class RecordWithProjectionKey(record: Record, projectionKey: String)
 
   object State {
-    val empty: State = State(Map.empty, Vector.empty, Instant.EPOCH, 0)
+    val empty: State = State(Map.empty, Vector.empty, Instant.EPOCH, 0, Vector.empty)
 
     def apply(records: immutable.IndexedSeq[Record]): State = {
       if (records.isEmpty) empty
       else empty.add(records)
     }
+
+    def withKey(projectionKey: String, recordsWithKey: immutable.IndexedSeq[RecordWithProjectionKey]): State = {
+      val state = State(recordsWithKey.map(_.record))
+      val foreignLatestSortedByTimestamp = recordsWithKey
+        .groupBy(_.record.slice)
+        .map { case (_, records) => records.maxBy(_.record.timestamp) }
+        .filter(_.projectionKey != projectionKey)
+        .toVector
+        .sortBy(_.record.timestamp)
```

Comment:
Extra processing to find these, as the regular records in the state don't have the projection key.